frontend/app/src/composables/settings/api-keys/external/index.ts

```diff
@@ -0,0 +1,160 @@
+import { type MaybeRef } from '@vueuse/core';
+import {
+  type ExternalServiceKeys,
+  type ExternalServiceName
+} from '@/types/user';
+
+const getKey = (name: ExternalServiceName, chain?: string) => {
+  if (name === 'etherscan') {
+    assert(chain, 'chain is missing for etherscan');
+    if (chain === 'ethereum') {
+      return name;
+    }
+    return `${chain}_${name}`;
+  }
+  return name;
+};
+
+type Status = { message: string; success?: boolean };
+
+export const useExternalApiKeys = createSharedComposable(
+  (t: ReturnType<typeof useI18n>['t']) => {
+    const loading = ref(false);
+    const keys = ref<ExternalServiceKeys>();
+    const status = ref<Record<string, Status>>({});
+
+    const { show } = useConfirmStore();
+    const {
+      setExternalServices,
+      deleteExternalServices,
+      queryExternalServices
+    } = useExternalServicesApi();
+
+    const apiKey = (
+      name: MaybeRef<ExternalServiceName>,
+      chain?: MaybeRef<string>
+    ): ComputedRef<string> =>
+      computed(() => {
+        const items = get(keys);
+        const service = get(name);
+        if (!items) {
+          return '';
+        } else if (service === 'etherscan') {
+          const chainId = get(chain);
+          assert(chainId, 'missing chain for etherscan');
+          return items[service]?.[chainId]?.apiKey || '';
```

Comment:
need to convert this to camelcase since the key would be most probably already converted.